astropy/table/table.py

```diff
@@ -219,9 +220,24 @@ def __init__(self, data=None, masked=None, names=None, dtype=None,
         # Set up a placeholder empty table
         self._set_masked(masked)
         self.columns = self.TableColumns()
-        self.meta = meta
         self.formatter = self.TableFormatter()
 
+        # Validate input for meta
+        if meta is None or isinstance(meta, collections.Mapping):
+            self.meta = meta
+        else:
+            raise TypeError('Meta type {0} not allowed to init Table'
+                             .format(type(meta)))
+
+        # Validate input for names and dtype
+        if isinstance(names, six.string_types):
+            raise TypeError('Names type {0} not allowed to init Table'
+                             .format(type(names)))
+    
+        if isinstance(dtype, six.string_types):
+            raise TypeError('Dtype type {0} not allowed to init Table'
+                             .format(type(dtype)))
+
```

Comment:
This seems like overkill to me.  Also there are plenty of things that should be allowable for `dtype` other than strings.

I think as far the messages themselves it's less useful to tell users what isn't allowed "Type <bla> not allowed" than it is to tell them what they should do "meta argument must be a Mapping", etc.